pkg/driver/reaper.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"k8s.io/klog"
+)
+
+type reaper struct {
+	sigs   chan os.Signal
+	stopCh chan struct{}
+}
+
+func newReaper() *reaper {
+	sigs := make(chan os.Signal, 1)
+	stopCh := make(chan struct{})
+
+	signal.Notify(sigs, syscall.SIGCHLD)
+	return &reaper{
+		sigs:   sigs,
+		stopCh: stopCh,
+	}
+}
+
+// start starts the reaper
+func (r *reaper) start() {
+	go r.runLoop()
+}
+
+func (r *reaper) runLoop() {
+	for {
+		select {
+		case <-r.sigs:
+			var (
+				status syscall.WaitStatus
+				rusage syscall.Rusage
+			)
+			childPid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, &rusage)
+			if err != nil {
+				klog.Warningf("Failed to wait for child proces %s", err)
+			} else {
+				klog.V(4).Infof("Waited for child process %d", childPid)
```

Comment:
Does waiting on pid `-1`mean we are waiting for the child process to exit? Since this is waiting in a separate go-routine, is it just here to keep the parent process around? A short comment explaining what the runLoop does would be nice.